src/plugins/field_formats/common/converters/numeral.ts

```diff
@@ -31,12 +32,15 @@ export abstract class NumeralFormat extends FieldFormat {
     pattern: this.getConfig!(`format:${this.id}:defaultPattern`),
   });
 
-  protected getConvertedValue(val: number | string): string {
+  protected getConvertedValue(val: number | string | object): string {
     if (val === -Infinity) return '-∞';
     if (val === +Infinity) return '+∞';
-    if (typeof val !== 'number') {
+    if (typeof val === 'string') {
       val = parseFloat(val);
     }
+    if (typeof val === 'object') {
+      return JSON.stringify(val);
+    }
```

Comment:
yes, this makes lots of sense, tiny nit, unlikely, but theoretically it could be a `boolean` value, so handling it like this and it would work exactly like before for non objects
```suggestion
    if (typeof val === 'object') {
      return JSON.stringify(val);
    } else if (typeof val !== 'number') {
      val = parseFloat(val);
    }
```